2021-04-06 - Anton Farygin <rider@altlinux.org> 4.14.1-alt2
- fixed build with ocaml-4.12.0

--- xen-4.14.1/tools/ocaml/libs/xentoollog/xentoollog_stubs.c.orig	2020-12-17 16:47:25.000000000 +0000
+++ xen-4.14.1/tools/ocaml/libs/xentoollog/xentoollog_stubs.c	2021-03-01 22:53:23.106205483 +0000
@@ -53,8 +53,6 @@
 #include "_xtl_levels.inc"
 
 /* Option type support as per http://www.linux-nantes.org/~fmonnier/ocaml/ocaml-wrapping-c.php */
-#define Val_none Val_int(0)
-#define Some_val(v) Field(v,0)
 
 static value Val_some(value v)
 {
--- xen-4.14.1/tools/ocaml/libs/xl/xenlight_stubs.c.orig	2020-12-17 16:47:25.000000000 +0000
+++ xen-4.14.1/tools/ocaml/libs/xl/xenlight_stubs.c	2021-03-01 23:01:24.229859714 +0000
@@ -227,8 +227,6 @@
 }
 
 /* Option type support as per http://www.linux-nantes.org/~fmonnier/ocaml/ocaml-wrapping-c.php */
-#define Val_none Val_int(0)
-#define Some_val(v) Field(v,0)
 
 static value Val_some(value v)
 {
--- xen-4.14.1/tools/ocaml/libs/xc/xenctrl_stubs.c.orig	2020-12-17 16:47:25.000000000 +0000
+++ xen-4.14.1/tools/ocaml/libs/xc/xenctrl_stubs.c	2021-03-01 23:24:45.477290290 +0000
@@ -43,8 +43,6 @@
 #define _H(__h) ((xc_interface *)(__h))
 #define _D(__d) ((uint32_t)Int_val(__d))
 
-#define Val_none (Val_int(0))
-
 #define string_of_option_array(array, index) \
 	((Field(array, index) == Val_none) ? NULL : String_val(Field(Field(array, index), 0)))
 
